sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -729,6 +736,7 @@ def _migrate_rows(self) -> None:
                     for name in _parents_from_model(model, models)
                 )
             except Exception:
+                completed = False
```

Comment:
this is not correct, we can still migrate successfull, we just didn't auto convert everything

Reply:
So this progress should always be successful? I assumed this was the only "failure" state, but I can remove `completed` completely if needed.

Reply:
yea it's not necessary, the only failure state is if we rollback

Reply:
Another related detail here: do we want to bump the snapshot tracker in the progress bar if we get an exception? I was thinking that in the end of the migration it might be weird to see "Migration completed successfully" but have something like "2/3" in the batch column.

Reply:
> yea it's not necessary, the only failure state is if we rollback

I see, thanks for the explanation.

Reply:
well that could indicate how many snapshots were auto converted

Reply:
Cool, making the change now so that the migration is not marked as a success if we rollback.

Reply:
I added a `raise SQLMeshError(...)` after I ran `sqlmesh plan` the first time, so this is what it looks like when a migration fails (the 100% • 3/3 is expected here since I created the error here deliberately).

<img width="889" alt="Screenshot 2023-07-06 at 10 27 48 PM" src="https://github.com/TobikoData/sqlmesh/assets/46752250/751198e3-4f0e-455f-b0cb-34758bbe91fc">